blocks/product-gallery-block/features/product-gallery/default.jsx

```diff
@@ -0,0 +1,116 @@
+import React from "react";
+import PropTypes from "@arc-fusion/prop-types";
+import { useFusionContext, useComponentContext } from "fusion:context";
+import { RESIZER_APP_VERSION, RESIZER_URL } from "fusion:environment";
+import getProperties from "fusion:properties";
+import getTranslatedPhrases from "fusion:intl";
+
+import { Carousel, Icon, Image } from "@wpmedia/arc-themes-components";
+
+const BLOCK_CLASS_NAME = "b-product-gallery";
+
+export function ProductGalleryDisplay(props) {
+	const { arcSite, data, id, isFeaturedImageEnabled, resizerAppVersion, resizerURL } = props;
+
+	const { locale } = getProperties(arcSite);
+	const phrases = getTranslatedPhrases(locale);
+	const carouselItems = data?.schema?.productGallery?.value?.assets
+		.filter((asset) => asset.type === "image")
+		.slice(0, isFeaturedImageEnabled ? 9 : 8);
+
+	return (
+		<Carousel
+			className={`${BLOCK_CLASS_NAME}${
+				isFeaturedImageEnabled ? ` ${BLOCK_CLASS_NAME}--featured-image-enabled` : ""
+			}`}
+			key={id}
+			id={id}
+			label={phrases.t("product-gallery.aria-label")}
+			nextButton={
+				<Carousel.Button id={id} label={phrases.t("product-gallery.right-arrow-label")}>
+					<Icon name="ChevronRight" />
+				</Carousel.Button>
+			}
+			previousButton={
+				<Carousel.Button id={id} label={phrases.t("product-gallery.left-arrow-label")}>
+					<Icon name="ChevronLeft" />
+				</Carousel.Button>
+			}
+		>
+			{carouselItems?.map((item, carouselIndex) => {
+				const { url, auth, alt_text: altText, _id: itemId } = item;
+
+				// is this a featured image?
+				const isFeaturedImage = isFeaturedImageEnabled && carouselIndex === 0;
+
+				// take in app version that's the public key for the auth object in resizer v2
+				const targetAuth = auth[resizerAppVersion];
+				return (
+					<Carousel.Item
+						key={itemId}
+						label={`${phrases.t("product-gallery.slide-indicator", {
+							current: carouselIndex + 1,
+							maximum: carouselItems.length,
+						})}`}
+					>
+						<Image
+							alt={altText}
+							src={url}
+							resizedOptions={{ auth: targetAuth }}
+							width={372}
```

Comment:
Why choose 372 x 279 as the with and height values?

Reply:
from the mocks for mobile. would you have a different suggestion?

Reply:
375 was suggested for the base width and height from product